pkg/pipelines/backends.go

```diff
@@ -1,7 +1,13 @@
 package pipelines
 
+import corev1 "k8s.io/api/core/v1"
+
+type GenerationOpts struct {
+	ImagePullPolicy *corev1.PullPolicy
+}
+
 // Backend is an interface abstracting away the generator that will be used to generate manifests
 // for an execution environment.
 type Backend[T any] interface {
-	Generate() (T, error)
+	Generate(O GenerationOpts) (T, error)
```

Comment:
parameter names should not start with a capital letter. in this case you don't even need a parameter name since this is a method signature of an interface